Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -895,6 +895,7 @@ internal struct BinaryDecoder: Decoder {
 
         if let k = k, let v = v {
             value[k] = v
+            consumed = true
```

Comment:
@tbkka When looking at moving unknowns into the base Message, and always haven't them, so Maps tests started to failing, looking a little it seems like maps never say they consumed their input for the might have been ended up in both the unknowns and the real map field.  Does this fix seem correct?